proof/courier.go

```diff
@@ -467,6 +478,8 @@ func (h *HashMailBox) RecvAck(ctx context.Context, sid streamID) error {
 
 // CleanUp atempts to tear down the mailbox as specified by the passed sid.
 func (h *HashMailBox) CleanUp(ctx context.Context, sid streamID) error {
+	defer h.rawConn.Close()
```

Comment:
Non-blocking: I guess my request was less about actually logging the error but rather logging it instead of having it as a return value in the interface. Because swallowing the error here does show up as a code smell issue in my IDE (and will probably be flagged by the linter if we decide to turn on some of the rules we have in `lnd`).

Reply:
Maybe over-engineering:

In such use-cases we could try returning a named variable
```golang
func (h *HashMailBox) CleanUp(ctx context.Context, sid streamID) (err error) {
```
and assign to `err` the error we want to return within the body.
Then our `defer` would look like this
```golang
defer func(){
    if err != nil {
        err = h.rawConn.Close()
    }
}()
```

Reply:
Or yeah we could just omit returning `error`. Do we care about consistency on the `Close` func signature?